headers/components/component_names.hpp

```diff
@@ -26,6 +26,7 @@
 #define OH_COMPONENT_NAME   "OH"
 #define N2O_COMPONENT_NAME "N2O"
 #define TEMPERATURE_COMPONENT_NAME "temperature"
+#define TEMP_DOECLIM_COMPONENT_NAME "temp_doeclim"
```

Comment:
Should we use a more meaningful name?

Reply:
Ah, the "DOE" here doesn't refer to the department of energy. OK then.

Reply:
@bpbond and @rplzzz yea it's diffusive ocean energy climate model.  are there any licensing issues we need to be aware of of linking this one model with Hector? Can we still call it Hector?
https://github.com/scrim-network/BRICK

Reply:
We are using DOEClim with the permission of its authors, right?  If so, then it's fine.  We should, however, acknowledge the code they donated.  I have a note about that elsewhere.

Reply:
I have not done any investigations into permissions. DOECLIM has been used several times by Klaus Keller's group and I adapted their C++ implementation here. I'll check in with Ryan and Klaus about it.

Reply:
It looks like one or more versions of DOECLIM have been released under GPLv3.  Just get verbal confirmation from one of the original authors that they're ok with us releasing it under GPLv2.  (The differences aren't really relevant to what we're doing; I like v2 because it's shorter and written in plain language.)  Then we can just indicate that the code is used with permission.

Reply:
renamed component.  Still need to double check licensing.
(CH - tracking her changes)

Reply:
A short update on the permissions: we now have emailed permission from Elmar Kriegler (original creator of DOECLIM) to use DOECLIM in Hector under GPLv2. I'm waiting for a response from Klaus Keller (coauthor of the paper for which the GPLv3 version was written).

Reply:
We now also have written permission from the author of the DOECLIM version released under GPLv3